### PR TITLE
[PAY-1463] Fix d to set discovery node

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Fix the "d" dialog that changes discovery node by adding back the code that reads and sets the local storage cache for discovery node.

Web only.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cached-dn

File: packages/web/src/services/audius-sdk/discoveryNodeSelector.ts [hunk 1]
```diff
@@ -3,7 +3,70 @@ import { DiscoveryNodeSelectorService } from '@audius/common'
...
+type CachedDiscoveryNodeTimestamp =
+  | {
+      endpoint?: string
+      timestamp?: number
+    }
+  | null
+  | undefined
+
+const getCachedDiscoveryNode = () => {
+  const cached = localStorage.getItem(DISCOVERY_PROVIDER_TIMESTAMP)
```
Comment (piazzatron) [inline packages/web/src/services/audius-sdk/discoveryNodeSelector.ts:18]: nit: could use our `get/setExpiringJSONValue` methods from the LocalStorage class

File: packages/web/src/services/audius-sdk/discoveryNodeSelector.ts [hunk 2]
```diff
@@ -3,7 +3,70 @@ import { DiscoveryNodeSelectorService } from '@audius/common'
...
+      ) {
+        const cachedDiscoveryNode =
+          cachedDiscoveryNodeTimestamp.endpoint ?? undefined
+        console.debug(
+          '[discovery-node-selector-service] Using cached discovery node',
+          cachedDiscoveryNode
+        )
+        return cachedDiscoveryNode
+      } else {
+        console.debug(
```
Comment (piazzatron) [inline packages/web/src/services/audius-sdk/discoveryNodeSelector.ts:38]: Should we remove the localStorage entry in this case?

File: packages/web/src/services/audius-sdk/discoveryNodeSelector.ts [hunk 3]
```diff
@@ -3,7 +3,70 @@ import { DiscoveryNodeSelectorService } from '@audius/common'
...
+  if (cached) {
+    try {
+      const cachedDiscoveryNodeTimestamp = JSON.parse(
+        cached
+      ) as CachedDiscoveryNodeTimestamp
+      if (
+        cachedDiscoveryNodeTimestamp &&
+        cachedDiscoveryNodeTimestamp.timestamp &&
+        cachedDiscoveryNodeTimestamp.timestamp >
+          new Date().getTime() - CACHE_TTL
```
Comment (piazzatron) [inline packages/web/src/services/audius-sdk/discoveryNodeSelector.ts:28]:
```suggestion
      if (cachedDiscoveryNodeTimestamp?.timestamp > new Date().getTime() - CACHE_TTL
```

**Review by piazzatron — APPROVED** [review]
Thank you for doing this

File: packages/web/src/services/audius-sdk/discoveryNodeSelector.ts (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/services/audius-sdk/discoveryNodeSelector.ts:38]: It'll get overwritten by the listener

**Pull request merged**